qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -367,9 +367,73 @@ def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_acti
     return spinz_matrix_elements
 
 
+def get_particle_number_elements(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
+    r"""Reads the Hartree-Fock (HF) electronic structure data file, defines an active
+    space and generates the table required to build the particle number operator
+    :math:`\hat{N}=\sum_\alpha \hat{n}_\alpha`.
+
+    In general, if an active space is defined, the particle number operator reads,
+
+    .. math::
+
+        \hat{N} = 2 n_{\mathrm{docc}} + \sum_\alpha \hat{c}_\alpha^\dagger \hat{c}_\alpha,
+
+    where :math:`2 n_{\mathrm{docc}}` is the number of particles occupying
+    :math:`n_{\mathrm{docc}}` (frozen) core orbitals and :math:`\hat{c}_\alpha^\dagger`
+    (:math:`\hat{c}_\alpha`) is the creation (annihilation) particle operator acting on
+    the :math:`\alpha`-th active spin-orbital.
+
+    Args:
+        mol_name (str): name of the molecule
+        hf_data (str): path to the directory with the HF electronic structure data
+        n_active_electrons (int): number of active electrons
+        n_active_orbitals (int): number of active orbitals
+
+    Returns:
+
+        tuple: NumPy array with the table used to build the fermionic operator and
+        the number of particles occupying the core orbitals. The first two columns of
+        the table contain the index :math:`\alpha` and, in this particular case for :math:`\hat{N}`,
+        the entries of the third column are all equal to one.
+
+    **Example**
+
+    >>> pn_table, pn_docc = get_particle_number_table(
```

Comment:
Note the use of `_table` here whereas the function uses `_elements`. I prefer `_table` for consistency with other functions